solutions/USACO open16-diamond-bronze.mdx

```diff
@@ -0,0 +1,55 @@
+---
+id: usaco-639
+source: USACO Bronze 2016 Open
+title: Diamond Collector
+author: Danh Ta Chi Thanh
+---
+
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_diamond_bronze_open16.html)
+
+
+
+<LanguageSection>
+
+<CPPSection>
+
+## C++ Implementation
+
+```cpp
+#include <iostream>
+#include <cstdio>
+#include <algorithm>
+#include <vector>
+
+using namespace std;
+
+int main(){
+	ios_base::sync_with_stdio(0); cin.tie(0);
+	freopen("diamond.in","r",stdin);
+	freopen("diamond.out","w",stdout);
+	int n,k;
+	cin >> n >> k;
+	vector<int> a;
+	a.resize(n);
+	for (int i=0;i<n;++i) {
+		cin >> a[i];
+	}
+	sort(a.begin(),a.end());
+	int ans=0;
+	for (int i=0;i<n;++i) {
+		int cnt=1;
+		for (int j=i+1;j<n;++j) {
+			if (a[j]-a[i]<=k) ++cnt;
+			else break;
+		}
+		ans=max(cnt,ans);
+	}
+	cout << ans;
+	return 0;
+}
+
+```
+
+</CPPSection>
+

```

Comment:
Looks good! I personally prefer to put one space after every ; in the for loop, also I like to put spaces in variable declarations. But that is personal preference and probably not necessary, another reviewer will tell you if is ok.